### DISCUSSION TITLE
Allow aborting inspector value changes with right mouse button or escape

### DISCUSSION THREAD

**Discussion by passivestar** [godotengine/godot-proposals#9044] [Editor]
When moving/rotating/scaling objects using gizmos in the 3d viewport it's possible to right click or press escape to abort before you release the left mouse button. But it's currently not the case when dragging values in inspector. In Blender you can do both

It's often convenient to see what effect changing a certain value has without actually committing to the change, and pressing right mouse button is a fast and convenient way of doing that, as it's less friction than using Ctrl+Z

_(Here's Blender doing the thing on RMB):_

https://github.com/godotengine/godot-proposals/assets/60579014/fda37bad-4b3a-4351-b17f-de8014324ef3

**Comment by lostminds** [comment]
I think this could be a good idea, but there are two different ideas here that I think might be considered separately, even if they have similar use. And your current title might confuse people as to which one you mean:
- The first (that I thought you meant) is to be able to cancel ongoing edits in inspector value fields by hitting escape to revert their content. This would be nice, but for this case right-click makes no sense.
- The second (that I now think you meant) is to be able to cancel and revert mouse drag-operations in the main viewport by clicking right mouse or esc. This could also be useful, but would probably be easier to understand if you didn't call it changing values in the inspector (the right side panel).

**Reply by passivestar** [reply]
The second is already possible in godot, which is what I meant when I wrote

> in the 3d viewport it's possible to right click or press escape to abort before you release the left mouse button

And the first is what I'm proposing. Both escape and right click work in blender (in the video I'm showing it with right click), and right click is actually more convenient than escape because it's right under your finger at all times 🙂

**Reply by lostminds** [reply]
Right, I see now, sorry for adding more confusion then. I didn't know the viewport right-click cancel was already a thing and was confused since that's what the video showed first. Then it makes a lot of sense to have the same right click cancel when mouse drag-editing values in the inspector value fields.

**Comment by passivestar** [comment]
PR here: https://github.com/godotengine/godot/pull/88275

**Reply by Calinou** [reply]
That PR is now merged, closing. Thanks for contributing this change :slightly_smiling_face:
